app/dao/author_dao.py

```diff
@@ -1,28 +1,41 @@
 from app.helpers.decorators import is_not_null
 from app.model.author import Author
+from app.config.setup_db import Connection
 
 from typing import List
 
 
 class AuthorDao:
     """Banco de Dados de Autores"""
 
-    list_authors: List[Author] = []
+    def __init__(self):
+        self.connection = Connection.connect()
+        self.db = self.connection.cursor(dictionary=True)
 
     def save(self, author: Author) -> None:
         if not isinstance(author, Author):
             raise TypeError('O argumento passado não é do tipo Autor')
 
-        if author in self.list_authors:
-            raise ValueError('Este e-mail já consta na base de dados.')
+        sql = 'INSERT INTO author(name, email) VALUES (%s, %s)'
+        args = (author.name, author.email)
 
-        self.list_authors.append(author)
+        self.db.execute(sql, args)
+        self.connection.commit()
 
-        print(f'\n---Autor Cadastrado---\n{author}')
+        print(f'\n---Autor inserido no banco---\n{author}')
 
     @is_not_null
     def find_one(self, email: str) -> Author:
-        for author in self.list_authors:
-            if email.lower() == author.email.lower():
-                return author
+        sql = 'SELECT name, email FROM author WHERE email = %s'
+        self.db.execute(sql, (email, ))
+
+        result_query = self.db.fetchone()
+
+        if result_query:
+            return Author(result_query['name'], result_query['email'])
+
         raise KeyError('Este autor não está cadastrado')
+
+    def close(self):
```

Comment:
Esse trecho vai ficar repetido em todos seus DAOs?

Reply:
É porque não sei a hora que o usuário irá acabar de usar as classes, logo, pensei que ao acabar, ele mesmo poderia fechar. Mas tudo bem, posso usar um gerenciador de contexto para isso.

Reply:
Aqui tem duas questões: 

1 - Se você não sabe, mas dá a conexão implicitamente pra ele, como ele vai saber que precisa chamar o close no final? 
2 - Se ele mesmo poderia fechar, por que ele mesmo não abre também? (Inversão de Controle). 

A maior discussão é: se uma parte do comportamento é implícita, por que a outra não é? As duas coisas estão muito amarradas, a abertura e fechamento da conexão, as duas devem seguir o mesmo padrão, se a abertura é implícita o fechamento também deve ser, o mesmo vale se for explicito. 

Por questão de experiência, eu sugiro você deixar tudo implícito para você sentir a dor que isso pode te gerar cards mais a frente 😺 #nãoPassará!